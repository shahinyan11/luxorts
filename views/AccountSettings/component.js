import { Button, Dropdown } from 'antd';
import { FormattedMessage } from 'react-intl';

import { MENU_ITEMS } from 'constants/accountSettings';
import { ACCEPT_AVATAR_TYPES } from 'constants/validations';

import getUserFullName from 'utils/users/getUserFullName';

import SvgIcon from 'views/shared/SvgIcon';
import MenuItem from 'views/AccountSettings/MenuItem';
import AccountSettingsLayout from 'views/layouts/AccountSettings';
import UserAvatar from 'views/shared/UserAvatar';

import useContainer, { getInitialProps } from './hook';

const AccountSettings = () => {
  const {
    handleUploadAvatarClick,
    handleRemoveAvatarClick,
    handleChangeFile,
    handleChangeAvatarImage,
    currentUser,
    avatarImgUrl,
    inputFileRef,
  } = useContainer();

  return (
    <>
      <input
        ref={inputFileRef}
        className="d-none"
        type="file"
        accept={ACCEPT_AVATAR_TYPES}
        onChange={handleChangeFile}
      />
      <div className="account">
        <div className="d-flex justify-content-space-between flex-column flex-sm-row align-items-center mb-40 mb-lg-80">
          <h1 className="text-display-2 mb-16 mb-sm-0">
            <FormattedMessage id="accountSettings.title" />
          </h1>
        </div>
        <div className="account__user-header d-flex flex-column align-items-center mb-64">
          <div className="account__user-avatar mb-16">
            <UserAvatar size="xxxl" currentUser={currentUser} />
            {avatarImgUrl && (
              <Dropdown
                overlay={
                  <ul className="account-controls">
                    <li className="account-controls__item">
                      <a
                        role="button"
                        className="account-controls__link d-block w-100"
                        onClick={handleChangeAvatarImage}
                      >
                        <FormattedMessage id="shared.change" />
                      </a>
                    </li>
                    <li className="account-controls__item">
                      <a
                        role="button"
                        className="account-controls__link d-block w-100"
                        onClick={handleRemoveAvatarClick}
                      >
                        <FormattedMessage id="shared.remove" />
                      </a>
                    </li>
                  </ul>
                }
                placement="bottomCenter"
              >
                <Button htmlType="button" className="account__user-upload">
                  <SvgIcon icon="camera" className="account__user-icon" />
                </Button>
              </Dropdown>
            )}
            {!avatarImgUrl && (
              <Button
                htmlType="button"
                className="account__user-upload"
                onClick={handleUploadAvatarClick}
              >
                <SvgIcon icon="camera" className="account__user-icon" />
              </Button>
            )}
          </div>
          <p className="account__user-name mb-4 text-headline-2">{getUserFullName(currentUser)}</p>
          <p className="account__user-mail mb-0 text-body">{currentUser?.email}</p>
        </div>
      </div>
      <ul className="account-menu mb-40 mb-md-80 mb-lg-160">
        {MENU_ITEMS.map((data) => (
          <MenuItem key={data.title.id} {...data} />
        ))}
      </ul>
    </>
  );
};

AccountSettings.getInitialProps = getInitialProps;

AccountSettings.Layout = AccountSettingsLayout;

export default AccountSettings;
