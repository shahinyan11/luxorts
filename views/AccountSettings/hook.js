import { useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { path } from 'ramda';

import useImmutableCallback from 'hooks/useImmutableCallback';

import { showModal } from 'state/modal/actions';
import { currentUserSelector } from 'state/concepts/session/selectors';

import checkUserVerification from 'utils/auth/checkUserVerification';

function useContainer() {
  const inputFileRef = useRef();
  const dispatch = useDispatch();
  const currentUser = useSelector(currentUserSelector);
  const avatarImgUrl = path(['userProfile', 'avatarUrls', 'original'], currentUser);

  /**
   * Show UPLOAD_PROFILE_IMAGE_MODAL
   * @type {(function(*): void)|*}
   */
  const handleUploadAvatarClick = useImmutableCallback(() => {
    dispatch(
      showModal({
        modalType: 'UPLOAD_PROFILE_IMAGE_MODAL',
      }),
    );
  });

  /**
   * Show REMOVE_PHOTO_MODAL
   * @type {(function(*): void)|*}
   */
  const handleRemoveAvatarClick = useImmutableCallback(() => {
    dispatch(
      showModal({
        modalType: 'REMOVE_PHOTO_MODAL',
      }),
    );
  });

  /**
   * Change avatar image
   * @param event {object}
   * @type {(function(*): void)|*}
   */
  const handleChangeAvatarImage = useImmutableCallback((event) => {
    event.stopPropagation();

    inputFileRef.current.click();
  });

  /**
   * Handle change image avatar file
   * @param event {object}
   * @type {(function(*): void)|*}
   */
  const handleChangeFile = useImmutableCallback((event) => {
    const file = event.target.files[0];

    if (!file) {
      return;
    }

    dispatch(
      showModal({
        modalType: 'UPLOAD_PROFILE_IMAGE_MODAL',
        modalProps: {
          initialImgSrc: URL.createObjectURL(file),
        },
      }),
    );

    // clear input file
    inputFileRef.current.value = null;
  });

  return {
    currentUser,
    handleUploadAvatarClick,
    handleRemoveAvatarClick,
    handleChangeAvatarImage,
    handleChangeFile,
    avatarImgUrl,
    inputFileRef,
  };
}

export const getInitialProps = async (ctx) => {
  checkUserVerification(ctx);
};

export default useContainer;
