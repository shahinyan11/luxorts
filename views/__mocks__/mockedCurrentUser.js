export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
