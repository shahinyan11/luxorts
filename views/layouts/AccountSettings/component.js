import PropTypes from 'prop-types';

import Header from 'views/shared/headers/Header';
import Footer from 'views/shared/footers/Footer';

const AccountSettingsLayout = ({ children }) => (
  <div className="main-layout">
    <Header />
    <div className="main-layout__content">{children}</div>
    <Footer isAccount />
  </div>
);

AccountSettingsLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default AccountSettingsLayout;
