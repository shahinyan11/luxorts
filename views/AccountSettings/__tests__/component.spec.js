import { shallow } from 'enzyme';

import mockedCurrentUser from 'views/__mocks__/mockedCurrentUser';

import useContainer from '../hook';
import AccountSettings from '../component';

const mockedHookData = {
  currentUser: mockedCurrentUser,
  handleUploadAvatarClick: jest.fn(),
  handleRemoveAvatarClick: jest.fn(),
  handleChangeFile: jest.fn(),
  handleChangeAvatarImage: jest.fn(),
  avatarImgUrl: 'avatarImgUrl',
  inputFileRef: jest.fn(),
};

jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('AccountSettings component', () => {
  let component = shallow(<AccountSettings />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when avatarImgUrl isn`t present', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      avatarImgUrl: null,
    });

    component = shallow(<AccountSettings />);

    expect(component).toMatchSnapshot();
  });
});
