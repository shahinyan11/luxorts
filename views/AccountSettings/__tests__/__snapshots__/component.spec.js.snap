// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSettings component matches snapshot 1`] = `
<Fragment>
  <input
    accept="image/jpeg,image/jpg,image/png"
    className="d-none"
    onChange={[MockFunction]}
    type="file"
  />
  <div
    className="account"
  >
    <div
      className="d-flex justify-content-space-between flex-column flex-sm-row align-items-center mb-40 mb-lg-80"
    >
      <h1
        className="text-display-2 mb-16 mb-sm-0"
      >
        <Memo(FormattedMessage)
          id="accountSettings.title"
        />
      </h1>
    </div>
    <div
      className="account__user-header d-flex flex-column align-items-center mb-64"
    >
      <div
        className="account__user-avatar mb-16"
      >
        <UserAvatar
          currentUser={
            Object {
              "email": "evhenii.hryhoriev+10@rubygarage.org",
              "id": "5c5d814c-8e06-433d-b006-4119311585bd",
              "publicId": "c4toxHVi",
              "userInvitation": Array [
                Object {
                  "created_at": "2022-06-23T06:32:55.588Z",
                  "email": "12@mail.ru",
                  "id": "52080e19-c5e0-45bb-af58-edbbe70e78fa",
                  "last_sent_at": "2022-06-23T06:32:55.474Z",
                  "resendable": true,
                  "status": "pending",
                  "user_account_id": "7211ccc2-1fca-4799-b1e1-4005e9fc42be",
                },
              ],
              "userProfile": Object {
                "about": null,
                "address": null,
                "avatarUrls": Object {
                  "original": "/avatar-img-url",
                },
                "city": "city",
                "country": "country",
                "dateOfBirth": "1994-01-01",
                "drivingLicenseFirstName": "Evhenii",
                "drivingLicenseLastName": "Test",
                "firstName": "Evhenii",
                "gender": null,
                "id": "52080e19-c5e0-45bb-af58-edbbe70e78fa",
                "lastName": "Test",
                "phoneNumber": "+299999999",
                "state": "state",
                "street": "street",
                "zipCode": "zipCode",
              },
              "userType": "traveller_account",
              "verifiedStatus": "unverified",
            }
          }
          size="xxxl"
        />
        <Dropdown
          mouseEnterDelay={0.15}
          mouseLeaveDelay={0.1}
          overlay={
            <ul
              className="account-controls"
            >
              <li
                className="account-controls__item"
              >
                <a
                  className="account-controls__link d-block w-100"
                  onClick={[MockFunction]}
                  role="button"
                >
                  <Memo(FormattedMessage)
                    id="shared.change"
                  />
                </a>
              </li>
              <li
                className="account-controls__item"
              >
                <a
                  className="account-controls__link d-block w-100"
                  onClick={[MockFunction]}
                  role="button"
                >
                  <Memo(FormattedMessage)
                    id="shared.remove"
                  />
                </a>
              </li>
            </ul>
          }
          placement="bottomCenter"
        >
          <Button
            className="account__user-upload"
            htmlType="button"
          >
            <SvgIcon
              className="account__user-icon"
              icon="camera"
            />
          </Button>
        </Dropdown>
      </div>
      <p
        className="account__user-name mb-4 text-headline-2"
      >
        Evhenii Test
      </p>
      <p
        className="account__user-mail mb-0 text-body"
      >
        evhenii.hryhoriev+10@rubygarage.org
      </p>
    </div>
  </div>
  <ul
    className="account-menu mb-40 mb-md-80 mb-lg-160"
  >
    <MenuItem
      description={
        Object {
          "id": "shared.providePersonalDetailsAndHowWeCanReachYou",
        }
      }
      href="/account-settings/personal-information"
      icon="profile"
      key="personalInformation.title"
      title={
        Object {
          "id": "personalInformation.title",
        }
      }
    />
    <MenuItem
      description={
        Object {
          "id": "shared.updateYourPasswordAndSecureYourAccount",
        }
      }
      href="/account-settings/login-and-security"
      icon="user-shield"
      key="shared.loginSecurity"
      title={
        Object {
          "id": "shared.loginSecurity",
        }
      }
    />
    <MenuItem
      description={
        Object {
          "id": "shared.reviewPaymentsPayouts",
        }
      }
      href="#"
      icon="bank-card"
      key="shared.paymentsPayouts"
      title={
        Object {
          "id": "shared.paymentsPayouts",
        }
      }
    />
    <MenuItem
      description={
        Object {
          "id": "shared.inviteFriendsToJoinLuxortsCommunity",
        }
      }
      href="/account-settings/invite-friends"
      icon="guests"
      key="shared.inviteFriends"
      title={
        Object {
          "id": "shared.inviteFriends",
        }
      }
    />
  </ul>
</Fragment>
`;

exports[`AccountSettings component matches snapshot when avatarImgUrl isn\`t present 1`] = `
<Fragment>
  <input
    accept="image/jpeg,image/jpg,image/png"
    className="d-none"
    onChange={[MockFunction]}
    type="file"
  />
  <div
    className="account"
  >
    <div
      className="d-flex justify-content-space-between flex-column flex-sm-row align-items-center mb-40 mb-lg-80"
    >
      <h1
        className="text-display-2 mb-16 mb-sm-0"
      >
        <Memo(FormattedMessage)
          id="accountSettings.title"
        />
      </h1>
    </div>
    <div
      className="account__user-header d-flex flex-column align-items-center mb-64"
    >
      <div
        className="account__user-avatar mb-16"
      >
        <UserAvatar
          currentUser={
            Object {
              "email": "evhenii.hryhoriev+10@rubygarage.org",
              "id": "5c5d814c-8e06-433d-b006-4119311585bd",
              "publicId": "c4toxHVi",
              "userInvitation": Array [
                Object {
                  "created_at": "2022-06-23T06:32:55.588Z",
                  "email": "12@mail.ru",
                  "id": "52080e19-c5e0-45bb-af58-edbbe70e78fa",
                  "last_sent_at": "2022-06-23T06:32:55.474Z",
                  "resendable": true,
                  "status": "pending",
                  "user_account_id": "7211ccc2-1fca-4799-b1e1-4005e9fc42be",
                },
              ],
              "userProfile": Object {
                "about": null,
                "address": null,
                "avatarUrls": Object {
                  "original": "/avatar-img-url",
                },
                "city": "city",
                "country": "country",
                "dateOfBirth": "1994-01-01",
                "drivingLicenseFirstName": "Evhenii",
                "drivingLicenseLastName": "Test",
                "firstName": "Evhenii",
                "gender": null,
                "id": "52080e19-c5e0-45bb-af58-edbbe70e78fa",
                "lastName": "Test",
                "phoneNumber": "+299999999",
                "state": "state",
                "street": "street",
                "zipCode": "zipCode",
              },
              "userType": "traveller_account",
              "verifiedStatus": "unverified",
            }
          }
          size="xxxl"
        />
        <Button
          className="account__user-upload"
          htmlType="button"
          onClick={[MockFunction]}
        >
          <SvgIcon
            className="account__user-icon"
            icon="camera"
          />
        </Button>
      </div>
      <p
        className="account__user-name mb-4 text-headline-2"
      >
        Evhenii Test
      </p>
      <p
        className="account__user-mail mb-0 text-body"
      >
        evhenii.hryhoriev+10@rubygarage.org
      </p>
    </div>
  </div>
  <ul
    className="account-menu mb-40 mb-md-80 mb-lg-160"
  >
    <MenuItem
      description={
        Object {
          "id": "shared.providePersonalDetailsAndHowWeCanReachYou",
        }
      }
      href="/account-settings/personal-information"
      icon="profile"
      key="personalInformation.title"
      title={
        Object {
          "id": "personalInformation.title",
        }
      }
    />
    <MenuItem
      description={
        Object {
          "id": "shared.updateYourPasswordAndSecureYourAccount",
        }
      }
      href="/account-settings/login-and-security"
      icon="user-shield"
      key="shared.loginSecurity"
      title={
        Object {
          "id": "shared.loginSecurity",
        }
      }
    />
    <MenuItem
      description={
        Object {
          "id": "shared.reviewPaymentsPayouts",
        }
      }
      href="#"
      icon="bank-card"
      key="shared.paymentsPayouts"
      title={
        Object {
          "id": "shared.paymentsPayouts",
        }
      }
    />
    <MenuItem
      description={
        Object {
          "id": "shared.inviteFriendsToJoinLuxortsCommunity",
        }
      }
      href="/account-settings/invite-friends"
      icon="guests"
      key="shared.inviteFriends"
      title={
        Object {
          "id": "shared.inviteFriends",
        }
      }
    />
  </ul>
</Fragment>
`;
